core/src/mender-client.c

```diff
@@ -386,117 +349,49 @@ mender_client_activate(void) {
     }
 #endif /* CONFIG_MENDER_CLIENT_INVENTORY */
 
-END:
-
-    return ret;
-}
-
-mender_err_t
-mender_client_deactivate(void) {
-
-    mender_err_t ret;
-
-#ifdef CONFIG_MENDER_CLIENT_INVENTORY
-    if (MENDER_OK != (ret = mender_inventory_deactivate())) {
-        mender_log_error("Unable to deactivate the inventory functionality");
-        /* keep going on, we want to do as much cleanup as possible */
-    }
-#endif /* CONFIG_MENDER_CLIENT_INVENTORY */
```

Comment:
`mender_inventory_activate()` is kept in place above. I think we should keep the deactivation here too. We require no inventory in builds so this should not be a functional problem. I don't see a reason to throw this code away.

Reply:
Right. Thank you.

Reply:
Okey. But the issue is that the caller `mender_client_deactivate` is not kept. I could keep `mender_client_deactivate` but calling it will be undefined behavior.

I think is best to remove the whole thing as we need to re-implement with the new scheduler on [MEN-7536](https://northerntech.atlassian.net/browse/MEN-7536)